src/cargo/ops/resolve.rs

```diff
@@ -316,9 +317,19 @@ fn add_overrides<'a>(registry: &mut PackageRegistry<'a>,
 }
 
 fn get_resolved_packages<'a>(resolve: &Resolve,
-                             registry: PackageRegistry<'a>)
+                             registry: PackageRegistry<'a>,
+                             excludes: &str)
                              -> PackageSet<'a> {
-    let ids: Vec<PackageId> = resolve.iter().cloned().collect();
+    let excludes = excludes.split(",").collect::<Vec<&str>>();
+    let filter = |id: &PackageId| {
+        for e in &excludes {
+            if &id.name() == e {
+                return false;
+            }
+        }
+        true
+    };
+    let ids: Vec<PackageId> = resolve.iter().cloned().filter(|id| filter(id)).collect();
```

Comment:
This can be rewritten as:

    filter(filter)

And then we should rename the `filter` predicate.